### PR TITLE
Add db845c machine and kernel

### PR DESCRIPTION
**Pull request by alimon**

### CODE REVIEW

File: conf/machine/include/qcom-sdm845.inc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+MACHINE_ESSENTIAL_EXTRA_RRECOMMENDS += " \
+"
+
+# Fastboot expects an ext4 image, which needs to be 4096 aligned
+IMAGE_FSTYPES ?= "ext4.gz"
+IMAGE_ROOTFS_ALIGNMENT = "4096"
+
+QCOM_BOOTIMG_KERNEL_BASE ?= "0x80000000"
+QCOM_BOOTIMG_PAGE_SIZE ?= "4096"
```
[the remark this answers is not in the document]
Comment (alimon) [reply conf/machine/include/qcom-sdm845.inc]: @andersson Yes it works but if is better to set 2048, I can change it.

[the remark this answers is not in the document]
Comment (alimon) [reply conf/machine/include/qcom-sdm845.inc]: @andersson I changed to use 2048 page size..

[the remark this answers is not in the document]
Comment (ndechesne) [reply conf/machine/include/qcom-sdm845.inc]: I thought it had to be a multiple of the physical page size on the target device (e.g. 4096). on DB820c we found issues when using 2048 (see https://git.linaro.org/ci/job/configs.git/commit/?id=5b15990b88f35b3db4eb8776b9423b08d90fde61, though it sadly doesn't say what the issue was...). 

mkbootimg will pad all sections (kernel, ramdisk, ..) to PAGE_SIZE, and we ended up with an image that wouldn't boot on 820c, if 'something' wasn't aligned well. doesn't that ring a bell?

[the remark this answers is not in the document]
Comment (andersson) [reply conf/machine/include/qcom-sdm845.inc]: I use 2k for 404, 410 and 845 and 4k for 820 and 835, so I'm not sure there is a pattern here. But iirc the only one where 4k doesn't work is 410, so how about we just go with 4k as standard and make an exception for the one that needs 2k?

[the remark this answers is not in the document]
Comment (alimon) [reply conf/machine/include/qcom-sdm845.inc]: @ndechesne @andersson I set the pagesize to 4096.

File: conf/machine/dragonboard-845c.conf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+KERNEL_IMAGETYPE ?= "Image.gz"
+KERNEL_DEVICETREE ?= "qcom/sdm845-db845c.dtb"
+
+SERIAL_CONSOLE ?= "115200 ttyMSM0"
+
+MACHINE_ESSENTIAL_EXTRA_RRECOMMENDS += " \
+    kernel-modules \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'opengl', 'mesa-driver-msm', '', d)} \
+    ${@bb.utils.contains('DISTRO_FEATURES', 'wifi', '', '', d)} \
```
Comment (ndechesne) [inline conf/machine/dragonboard-845c.conf]: not needed.

File: conf/machine/include/qcom-sdm845.inc (hunk 1 above)
Comment (ndechesne) [inline conf/machine/include/qcom-sdm845.inc]: on 820c, we have this:

# UFS partitions setup with 4096 logical sector size
EXTRA_IMAGECMD_ext4 += " -b 4096 "

don't we need that here too? I forgot the exact reason why i added it for 820c..

File: conf/machine/dragonboard-845c.conf (hunk 2 above)
Comment (alimon) [reply conf/machine/dragonboard-845c.conf]: @ndechesne  ok removed.

File: conf/machine/include/qcom-sdm845.inc (hunk 1 above)
Comment (alimon) [reply conf/machine/include/qcom-sdm845.inc]: @ndechesne Right, I removed EXTRA_IMAGECMD_ext4 += " -b 4096 " but forget to remove IMAGE_ROOTFS_ALIGNMENT, fixed.

**Pull request merged**